src/context/ThemeContext.jsx

```diff
@@ -0,0 +1,44 @@
+import { createContext, useState, useEffect } from "react";
+
+const ThemeContext = createContext({
+  isDarkMode: false,
+  toggleTheme: () => {},
+});
+
+function ThemeContextProvider({ children }) {
+  let currentMode = localStorage.getItem("isDarkMode");
+  console.log(typeof currentMode);
+  if (currentMode == null) {
+    currentMode = "false";
+    localStorage.setItem("isDarkMode", "false");
+  } else {
+    if (currentMode == "false") {
+      document.body.classList.remove("dark-theme");
+      localStorage.setItem("isDarkMode", "false");
```

Comment:
redundant local Storage update : updating local Storage multiple times within the render phase, which is unnecessary